opennms-webapp/src/main/webapp/includes/grafana-box.jsp

```diff
@@ -0,0 +1,130 @@
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+--%>
+
+<%--
+  This page is included by other JSPs to create a box containing a list of grafana dashboards.
+
+  It expects that a <base> tag has been set in the including page
+  that directs all URLs to be relative to the servlet context.
+--%>
+
+<%@page language="java"
+        contentType="text/html"
+        session="true"
+        import="org.apache.commons.io.IOUtils,
+                org.apache.http.HttpHost,
+                org.apache.http.HttpResponse,
+                org.apache.http.client.methods.HttpGet,
+                org.apache.http.conn.ConnectTimeoutException,
+                org.apache.http.impl.client.DefaultHttpClient,
+                org.apache.http.params.HttpConnectionParams,
+                org.apache.http.params.HttpParams,
+                org.apache.http.util.EntityUtils,
+                java.net.UnknownHostException" %>
+
+<%@taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>
+<%@taglib uri="http://java.sun.com/jsp/jstl/fmt" prefix="fmt" %>
+
+<%
+    String grafanaApiKey = System.getProperty("org.opennms.grafanaBox.apiKey", "");
+    String grafanaProtocol = System.getProperty("org.opennms.grafanaBox.protocol", "http");
+    String grafanaHostname = System.getProperty("org.opennms.grafanaBox.hostname", "localhost");
+    String grafanaTag = System.getProperty("org.opennms.grafanaBox.tag", "");
+    int grafanaPort = Integer.parseInt(System.getProperty("org.opennms.grafanaBox.port", "3000"));
+
+    String responseString = null;
+
+    if (!"".equals(grafanaApiKey)
+            && !"".equals(grafanaHostname)
+            && !"".equals(grafanaProtocol)
+            && ("http".equals(grafanaProtocol) || "https".equals(grafanaProtocol))) {
+        try {
+            DefaultHttpClient httpClient = new DefaultHttpClient();
+            HttpParams httpParams = httpClient.getParams();
+            /**
+             * Setting the timeouts to assure that the landing page will be loaded. Making the
+             * call via JS isn't possible due to CORS-related problems with the Grafana server.
+             */
+            HttpConnectionParams.setConnectionTimeout(httpParams, 500);
+            HttpConnectionParams.setSoTimeout(httpParams, 500);
+            HttpHost httpHost = new HttpHost(grafanaHostname, grafanaPort, grafanaProtocol);
+            HttpGet httpGet = new HttpGet("/api/search/");
+            httpGet.setHeader("Authorization", "Bearer " + grafanaApiKey);
+            HttpResponse httpResponse = httpClient.execute(httpHost, httpGet);
+            responseString = IOUtils.toString(httpResponse.getEntity().getContent(), "UTF-8");
+        } catch (UnknownHostException e) {
+            // do nothing
+        } catch (ConnectTimeoutException e) {
+            // do nothing
```

Comment:
I'd also like to see some kind of error handling. Catching and not doing anything is usually a bad idea. At least log it somewhere. 

You may also try multi-catch:

    try {
        throw new SomeException("Frühstück ist fertig.");
    } catch (SomeException | SomeOtherException | AndAnotherOne e) {
        LOG.error(e);
    }